src/shared/test/integration/inter-frame-communication-test.js

```diff
@@ -0,0 +1,243 @@
+import { delay } from '../../../test-util/wait';
+import { Bridge as BridgeOri } from '../../bridge';
+import { ListenerCollection as ListenerCollectionOri } from '../../listener-collection';
+import { PortFinder as PortFinderOri } from '../../port-finder';
+import { PortProvider as PortProviderOri } from '../../port-provider';
+import { isMessageEqual } from '../../port-util';
+
+describe('PortProvider-PortFinder-Bridge integration', () => {
+  let destroyables;
+
+  // These proxy classes are to keep track of instances that need to be destroyed.
+  class Bridge extends BridgeOri {
+    constructor() {
+      super();
+      destroyables.push(this);
+    }
+  }
+
+  class ListenerCollection extends ListenerCollectionOri {
+    constructor() {
+      super();
+      destroyables.push(this);
+    }
+    destroy() {
```

Comment:
Makes me think if `ListenerCollection#removeAll` should be renamed `ListenerCollection#destroy` and tag the class as destroyable.